### ISSUE TITLE
Adding a new module causes hupper/watchdog to blow up

### ISSUE THREAD

**Issue by digitalresistor** [Pylons/hupper#10]
I added a new module to my code base, and as soon as I updated my Pyramid projects `__init__.py` for `config.scan` all hell broke loose:

```
Serving on http://0.0.0.0:6543
Killing server with PID 83267.
Traceback (most recent call last):
  File "/Users/xistence/.ve/fulgent/lib/python3.5/site-packages/hupper/reloader.py", line 107, in run
    if not self._run_worker():
  File "/Users/xistence/.ve/fulgent/lib/python3.5/site-packages/hupper/reloader.py", line 161, in _run_worker
    self.monitor.add_path(path)
  File "/Users/xistence/.ve/fulgent/lib/python3.5/site-packages/hupper/reloader.py", line 43, in add_path
    self.monitor.add_path(p)
  File "/Users/xistence/.ve/fulgent/lib/python3.5/site-packages/hupper/watchdog.py", line 34, in add_path
    if dirpath not in self.dirpaths:
  File "/Users/xistence/.ve/fulgent/lib/python3.5/site-packages/watchdog/observers/fsevents.py", line 172, in schedule
    return BaseObserver.schedule(self, event_handler, path, recursive)
  File "/Users/xistence/.ve/fulgent/lib/python3.5/site-packages/watchdog/observers/api.py", line 290, in schedule
    timeout=self.timeout)
  File "/Users/xistence/.ve/fulgent/lib/python3.5/site-packages/watchdog/observers/fsevents.py", line 73, in __init__
    self.snapshot = DirectorySnapshot(watch.path, watch.is_recursive)
  File "/Users/xistence/.ve/fulgent/lib/python3.5/site-packages/watchdog/utils/dirsnapshot.py", line 207, in __init__
    st = stat(path)
FileNotFoundError: [Errno 2] No such file or directory: '/Users/xistence/Projects/fulgent/fulgent/subscribers/__pycache__'

During handling of the above exception, another exception occurred:

ValueError: PyCapsule_GetPointer called with invalid PyCapsule object

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/xistence/.ve/fulgent/bin/pserve", line 11, in <module>
    load_entry_point('pyramid==1.8.dev0', 'console_scripts', 'pserve')()
  File "/Users/xistence/.ve/fulgent/lib/python3.5/site-packages/pyramid/scripts/pserve.py", line 37, in main
    return command.run()
  File "/Users/xistence/.ve/fulgent/lib/python3.5/site-packages/pyramid/scripts/pserve.py", line 154, in run
    verbose=self.options.verbose,
  File "/Users/xistence/.ve/fulgent/lib/python3.5/site-packages/hupper/reloader.py", line 273, in start_reloader
    return reloader.run()
  File "/Users/xistence/.ve/fulgent/lib/python3.5/site-packages/hupper/reloader.py", line 113, in run
    self._stop_monitor()
  File "/Users/xistence/.ve/fulgent/lib/python3.5/site-packages/hupper/reloader.py", line 194, in _stop_monitor
    self.monitor.stop()
  File "/Users/xistence/.ve/fulgent/lib/python3.5/site-packages/hupper/reloader.py", line 49, in stop
    self.monitor.stop()
  File "/Users/xistence/.ve/fulgent/lib/python3.5/site-packages/watchdog/utils/__init__.py", line 99, in stop
    self.on_thread_stop()
  File "/Users/xistence/.ve/fulgent/lib/python3.5/site-packages/watchdog/observers/api.py", line 357, in on_thread_stop
    self.unschedule_all()
  File "/Users/xistence/.ve/fulgent/lib/python3.5/site-packages/watchdog/observers/api.py", line 353, in unschedule_all
    self._clear_emitters()
  File "/Users/xistence/.ve/fulgent/lib/python3.5/site-packages/watchdog/observers/api.py", line 231, in _clear_emitters
    emitter.stop()
  File "/Users/xistence/.ve/fulgent/lib/python3.5/site-packages/watchdog/utils/__init__.py", line 99, in stop
    self.on_thread_stop()
  File "/Users/xistence/.ve/fulgent/lib/python3.5/site-packages/watchdog/observers/fsevents.py", line 77, in on_thread_stop
    _fsevents.stop(self)
SystemError: <built-in function stop> returned a result with an error set
```

Steps:

1. Start `pserve` with `--reload`
2. `mkdir subscribers && touch subscribers/__init__.py`
    This didn't cause the reloader to do anything, because it didn't know about this path
3. Modify `__init__.py` to add `config.scan('.subscribers')`
4. Reloader reloads because of change to existing file
5. ???
6. Boom

**Comment by digitalresistor** [comment]
Hupper version: `hupper==0.3.6`

**Comment by mmerickel** [comment]
Can you confirm whether this only happens with watchdog?

**Comment by mmerickel** [comment]
I'm not having any luck reproducing this.

**Comment by mmerickel** [comment]
I managed to get this stacktrace when hitting ctrl-c on the process group. Specifically the second part about the PyCapsule pointer. Not the 'no such file or directory' part.

**Comment by digitalresistor** [comment]
It may be a race condition... `__pycache__` gets created when the code is imported, but if watchdog gets there first, things go boom.

I just am not sure how to solve the race problem.

**Comment by mmerickel** [comment]
> __pycache__ gets created when the code is imported, but if watchdog gets there first, things go boom.

I don't see why that would be an issue.

**Comment by mmerickel** [comment]
The second stack trace doesn't concern me. It looks like a bug in the shutdown code of the FSEvents (darwin) watchdog observer. Fine. The actual issue appears to be that watchdog is attempting to `stat` a non-existent folder and crashes. There are several guards in hupper to avoid watching non-existent folders, but some race condition is causing this to happen anyway. One option is to probably just catch the error, but in reality watchdog should be handling it better.

**Comment by mmerickel** [comment]
I added a possible fix for this in e29e926 (just squashing the error). Please let me know if you see the issue again.
